velodyne_driver/src/driver/driver.cc

```diff
@@ -93,88 +91,80 @@ VelodyneDriver::VelodyneDriver(ros::NodeHandle node,
   }
   else
   {
-    ROS_ERROR_STREAM("cut_angle parameter is out of range. Allowed range is "
-    << "between 0.0 and 2*PI or negative values to deactivate this feature.");
+    RCLCPP_ERROR(this->get_logger(),"cut_angle parameter is out of range."
+                 "Allowed range is between 0.0 and 2*PI or negative values to deactivate this feature.");
     cut_angle = -0.01;
   }
-
-  // Convert cut_angle from radian to one-hundredth degree, 
+  
+  // Convert cut_angle from radian to one-hundredth degree,
   // which is used in velodyne packets
   config_.cut_angle = int((cut_angle*360/(2*M_PI))*100);
 
   int udp_port;
-  private_nh.param("port", udp_port, (int) DATA_PORT_NUMBER);
-
-  // Initialize dynamic reconfigure
-  srv_ = boost::make_shared <dynamic_reconfigure::Server<velodyne_driver::
-    VelodyneNodeConfig> > (private_nh);
-  dynamic_reconfigure::Server<velodyne_driver::VelodyneNodeConfig>::
-    CallbackType f;
-  f = boost::bind (&VelodyneDriver::callback, this, _1, _2);
-  srv_->setCallback (f); // Set callback function und call initially
-
+  this->get_parameter_or("port", udp_port, (int) DATA_PORT_NUMBER);
+  
   // initialize diagnostics
   diagnostics_.setHardwareID(deviceName);
   const double diag_freq = packet_rate/config_.npackets;
   diag_max_freq_ = diag_freq;
   diag_min_freq_ = diag_freq;
-  ROS_INFO("expected frequency: %.3f (Hz)", diag_freq);
+  RCLCPP_INFO(this->get_logger(), "expected frequency: %.3f (Hz)", diag_freq);
 
   using namespace diagnostic_updater;
   diag_topic_.reset(new TopicDiagnostic("velodyne_packets", diagnostics_,
                                         FrequencyStatusParam(&diag_min_freq_,
                                                              &diag_max_freq_,
                                                              0.1, 10),
                                         TimeStampStatusParam()));
-  diag_timer_ = private_nh.createTimer(ros::Duration(0.2), &VelodyneDriver::diagTimerCallback,this);
-
+  diag_timer_  = this->create_wall_timer(200ms, std::bind(&VelodyneDriver::diagTimerCallback, this));
   // open Velodyne input device or file
   if (dump_file != "")                  // have PCAP file?
     {
       // read data from packet capture file
-      input_.reset(new velodyne_driver::InputPCAP(private_nh, udp_port,
+      input_.reset(new velodyne_driver::InputPCAP(shared_from_this() , udp_port,
                                                   packet_rate, dump_file));
     }
   else
     {
+      
       // read data from live socket
-      input_.reset(new velodyne_driver::InputSocket(private_nh, udp_port));
+      input_.reset(new velodyne_driver::InputSocket(shared_from_this() , udp_port));
     }
-
+  
   // raw packet output topic
   output_ =
-    node.advertise<velodyne_msgs::VelodyneScan>("velodyne_packets", 10);
+    this->create_publisher<velodyne_msgs::msg::VelodyneScan>("~/velodyne_packets", 10);
```

Comment:
This shouldn't have the `~/` in front. Just having it as `velodyne_packets` makes it possible to namespace it properly.